ecr-login/cache/file.go

```diff
@@ -98,6 +98,9 @@ func (f *fileCredentialCache) fullFilePath() string {
 // file access. There is not guarantee here for handling multiple writes at once since there is no out of process locking.
 func (f *fileCredentialCache) save(registryCache *RegistryCache) error {
 	defer log.Flush()
+	if _, err := os.Stat(f.path); err != nil {
```

Comment:
In the BuildCredentialsCache() it does rely on AWS_ECR_DISABLE_CACHE to map between the NullCache and the FileCache. I could see either adding a AWS_ECR_CACHE_DIR and defaulting to "~/.ecr" or passing additional parameters to the builder. I'd opt for the environment variable.

I think the moving the stat / mkdir to the builder makes sense to help fix failure cases earlier. If we really need it to be more flexible (in terms of API, we could extend the builder in the future).